lib/glific/erp.ex

```diff
@@ -113,33 +114,28 @@ defmodule Glific.ERP do
 
   @spec create_or_update_address(map(), String.t()) :: :ok | {:error, String.t()}
   defp create_or_update_address(registration, customer_name) do
-    billing_exists? = address_exists?(customer_name, "Billing")
-    permanent_exists? = address_exists?(customer_name, "Permanent/Registered")
-
     current_address = registration.org_details["current_address"]
     registered_address = registration.org_details["registered_address"]
     are_addresses_same = compare_addresses(current_address, registered_address)
 
-    case (if billing_exists? do
-            update_address(current_address, "Billing", customer_name)
-          else
-            create_address(current_address, "Billing", customer_name)
-          end) do
-      {:ok, _} -> :ok
+    with {:ok, _} <- handle_address(current_address, "Billing", customer_name),
+         {:ok, _} <-
+           if(are_addresses_same,
+             do: {:ok, nil},
+             else: handle_address(registered_address, "Permanent/Registered", customer_name)
+           ) do
```

Comment:
We could write the if else clause in `with` body instead as a `with` clause which can improve the readability. Also we can make the return type of `create_or_update_address` as `{:ok, any} | {:error, String.t()}` since upstream we never really care about the data in ok tuple.